pkg/member/participant.go

```diff
@@ -131,27 +146,36 @@ func (m *Participant) GetLeaderID() uint64 {
 }
 
 // GetLeader returns current leader of the election group.
-func (m *Participant) GetLeader() *tsopb.Participant {
+func (m *Participant) GetLeader() participant {
 	leader := m.leader.Load()
 	if leader == nil {
 		return nil
 	}
-	member := leader.(*tsopb.Participant)
+	member := leader.(participant)
 	if member.GetId() == 0 {
 		return nil
 	}
 	return member
 }
 
 // setLeader sets the member's leader.
-func (m *Participant) setLeader(member *tsopb.Participant) {
+func (m *Participant) setLeader(member participant) {
 	m.leader.Store(member)
 	m.lastLeaderUpdatedTime.Store(time.Now())
 }
 
 // unsetLeader unsets the member's leader.
 func (m *Participant) unsetLeader() {
-	m.leader.Store(&tsopb.Participant{})
+	var leader participant
+	switch m.serviceName {
```

Comment:
Will generic type eliminate these switches?

Reply:
Maybe rewrite it later.